web-bluetooth/gap-characteristics.js

```diff
@@ -0,0 +1,107 @@
+function onButtonClick() {
+  log('Requesting Bluetooth Device...');
+  navigator.bluetooth.requestDevice(
+    {filters: [{name: ''}], optionalServices: ['generic_access']})
+  .then(device => {
+    log('Connecting to GATT Server...');
+    return device.gatt.connect();
+  })
+  .then(server => {
+    log('Getting GAP Service...');
+    return server.getPrimaryService('generic_access');
+  })
+  .then(service => {
+    log('Getting GAP Characteristics...');
+    return service.getCharacteristics();
+  })
+  .then(characteristics => {
+    let queue = Promise.resolve();
+    characteristics.forEach(characteristic => {
+      switch (characteristic.uuid) {
+
+        case BluetoothUUID.getCharacteristic('gap.appearance'):
+          queue.then(readAppearanceValue(characteristic));
```

Comment:
Promises are tricky :sweat_smile: Here, you are calling `readAppearanceValue` immediately rather than waiting for the previous promise to resolve. You need to put the call to `readAppearanceValue` inside another function:
```js
queue.then(() => readAppearanceValue(characteristic));
```